ext/standard/var.c

```diff
@@ -145,14 +145,22 @@ PHPAPI void php_var_dump(zval *struc, int level) /* {{{ */
 			PUTS("}\n");
 			break;
 		case IS_OBJECT:
+			class_name = Z_OBJ_HANDLER_P(struc, get_class_name)(Z_OBJ_P(struc));
+
+			zend_class_entry *ce = Z_OBJCE_P(struc);
+			if (ce->ce_flags & ZEND_ACC_ENUM) {
+				zval *case_name_zval = zend_hash_index_find(ce->enum_pointer_table, (zend_long) Z_OBJ_P(struc));
```

Comment:
It seems like Z_OBJ_P would be prone to hash collisions, because `zend_array` effectively uses the integer `value % (2**n)` as a hash key (see zend_hash_index_find_bucket)?

`sizeof(zend_object)` is 56 bytes, and I assume would be aligned to 8 bytes at least for 64-bit alignment, so addresses % 8 would all be 0

- It'd work but would be inefficient for lookups when there were many cases
- Also, is it guaranteed that zend_long is at least the pointer size (not sure if that's actually a concern on any supported platforms, e.g. 64-bit or 48-bit pointers and 32-bit integers)

Z_OBJ_HANDLE (spl_object_id) would be less prone to collisions because it uses all integers, but be an extra pointer dereference

```
#define Z_OBJ_HANDLE(zval)          (Z_OBJ((zval)))->handle
```

Would it be practical to allocate extra space in the emalloc() call for zend_object just for cases, and use that to avoid a hash lookup (while keeping property count)?

Reply:
> Would it be practical to allocate extra space in the emalloc() call for zend_object just for cases, and use that to avoid a hash lookup (while keeping property count)?

I have also thought about this. Unfortunately I wasn't sure if there are assumptions about the zval object size and thus opted for this solution. I will try your suggestion.

Reply:
Z_OBJ_HANDLE is probably the most practical - it is only used for debug output right now, so changing zend_object_properties_size (Zend/zend_objects_API.h) doesn't seem worth it

There's probably a lot of edge cases and assumptions about the zend_object size that aren't worth changing if the performance impact isn't large

```
/* force ZEND_ACC_USE_GUARDS for all classes */
#define ZEND_COMPILE_GUARDS						(1<<10)
```

Reply:
> Z_OBJ_HANDLE (spl_object_id) would be less prone to collisions because it uses all integers, but be an extra pointer dereference

That probably doesn't matter too much. This is only called for `from`, `serialize` and `var_dump` all of which have plenty of other overhead to where this becomes completely negligible. Using the handle will also protect against moving the objects in memory. Thanks for your suggestion!

Reply:
We reverted back to `->case` so this is obsolete.